src/ui/otb/OtbRound.ts

```diff
@@ -98,6 +102,19 @@ export default class OtbRound implements OtbRoundInterface, PromotingInterface {
       this.replay.init(variant, initialFen, situations, ply)
     }
 
+    if (data.offlineClock) {
+      const clockType = data.offlineClock.clockType as ClockType
+      this.clock = clockSet[clockType](this.onFlag)
+      this.clock.setState(data.offlineClock)
+    }
+    else {
```

Comment:
I don't understand this block. If there is no `data.offlineClock`, you shouldn't instantiate clock.

Reply:
You are right.